variables.tf

```
@@ -1,4 +1,272 @@
-variable "example" {
-  description = "Example variable"
-  default     = "hello world"
+## eks_iam
+
+variable "iam_role_enabled" {
+  type        = bool
+  description = "Whether to create an IAM role. Setting this to `true` will also replace any occurrences of `{service_account_role_arn}` in `var.values_template_path` with the ARN of the IAM role created by this module."
+  default     = false
+}
+
+## eks_iam_policy
+
+variable "iam_source_json_url" {
+  type        = string
+  description = "IAM source json policy to download. This will be used as the `source_json` meaning the `var.iam_policy_statements` and `var.iam_policy_statements_template_path` can override it."
+  default     = null
+}
+
+variable "iam_policy_statements" {
+  type        = any
+  description = "IAM policy for the service account. Required if `var.iam_role_enabled` is `true`. This will not do variable replacements. Please see `var.iam_policy_statements_template_path`."
+  default     = {}
+}
+
+## eks_iam_role
+
+variable "aws_account_number" {
+  type        = string
+  description = "AWS account number of EKS cluster owner"
```

```suggestion
  description = "AWS account number of EKS cluster owner."
```

But also see comment regarding `aws` provider and `aws_partition` data source.